client/src/actions/hanabiActions.js

```diff
@@ -0,0 +1,30 @@
+export function resetGame(){
+  return {type: "RESET_GAME"}
+}
+
+export function addPlayer(playerName){  
+  return {type: "ADD_PLAYER", playerName}
+}
+
+export function startGame(originalPlayers, originalDeck){
+  const deck = [...originalDeck];
+  let cardCount = undefined;
+  if (originalPlayers.length > 1 && originalPlayers.length < 4){
+    cardCount = 5;
+  }else if (originalPlayers.length >= 4 && originalPlayers.length <= 5) {
+    cardCount = 4;
+  }
+  const players = originalPlayers.map(p => {
+    const hand = [];
+    for(let i=0; i < cardCount; i++ ){
+      hand.push(getRandCard(deck));
+    }
+    return Object.assign({}, p, { hand });
+  });
+  return {type: "START_GAME", players, deck}
+}
+
+const getRandCard = (deck) =>{
```

Comment:
since this is a first class citizen of JavaScript now that we defined the function as a variable, we should define this at the top of the script so it is used after it is defined. Otherwise we get a warning message.